azext_edge/edge/providers/orchestration/instances.py

```diff
@@ -0,0 +1,75 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+from typing import List, Optional
+
+from knack.log import get_logger
+from rich import print
+
+from ...util.az_client import get_iotops_mgmt_client, parse_resource_id
+from ...util.queryable import Queryable
+
+logger = get_logger(__name__)
+
+
+QUERIES = {
+    "get_cl_from_instance": """
+        resources
+        | where type =~ 'microsoft.extendedlocation/customlocations'
+        | where id =~ '{resource_id}'
+        | project id, name, properties
+        """
+}
+
+INSTANCES_API_VERSION = "2021-10-01-privatepreview"
+# TODO temporary
+BASE_URL = "https://eastus2euap.management.azure.com"
+QUALIFIED_RESOURCE_TYPE = "Private.IoTOperations/instances"
+
+
+class Instances(Queryable):
+    def __init__(self, cmd):
+        super().__init__(cmd=cmd)
+        self.iotops_mgmt_client = get_iotops_mgmt_client(
+            subscription_id=self.default_subscription_id, endpoint=BASE_URL, api_version=INSTANCES_API_VERSION
+        )
+
+    def show(self, name: str, resource_group_name: str, show_tree: Optional[bool] = None) -> Optional[dict]:
+        result = self.iotops_mgmt_client.instance.get(instance_name=name, resource_group_name=resource_group_name)
+
+        if show_tree:
+            self._show_tree(result)
+            return
+
+        return result
+
+    def list(self, resource_group_name: Optional[str] = None) -> List[dict]:
+        if resource_group_name:
+            return list(
+                self.iotops_mgmt_client.instance.list_by_resource_group(resource_group_name=resource_group_name)
+            )
+
+        return list(self.iotops_mgmt_client.instance.list_by_subscription())
```

Comment:
does the service not return a workable itterrable already?

Reply:
You mean the client? It is an iterable, but we don't currently need to process them, therefore the whole iterable chain is collected into a list and returned.

Reply:
yes - it is rare to see the iterable be converted to a list before it is returned, especially since there is no transformations/extra code modifying it

Reply:
looks like the function return type is Iterable[JSON]?

Reply:
Changed the method to return an iterable directly rather than convert to list - now we rely on cli core to unpack. The instance list test changed to unpack the iterable.